olive/strategy/search_parameter.py

```diff
@@ -130,21 +160,19 @@ def __init__(
         self.support = support
         self.default = default or self.get_invalid_choice()
 
-    def get_support(self) -> List[Any]:
-        raise NotImplementedError("Use get_support_with_args instead")
+    def get_parents(self) -> Tuple[str]:
+        return self.parents
 
-    def get_support_with_args(
-        self, parent_values: Dict[str, Any]
-    ) -> Union[List[str], List[int], List[float], List[bool]]:
+    def get_support(self, parent_values: Dict[str, Any] = None) -> Union[List[str], List[int], List[float], List[bool]]:
         """Get the support for the search parameter for a given parent value."""
-        # pylint: disable=arguments-differ
-        assert parent_values.keys() == set(self.parents), "parent values keys do not match the parents"
+        assert parent_values is not None
+        assert set(self.parents).issubset(parent_values.keys()), "Not all parents included in parent values keys"
```

Comment:
the error message doesn't sound correct. the condition is more like "no parents included in parent values keys"

Reply:
nevermind this is fine